fiftyone/__public__.py

```diff
@@ -5,16 +5,21 @@
 | `voxel51.com <https://voxel51.com/>`_
 |
 """
+import os as _os
 
 import fiftyone.core.config as _foc
-import fiftyone.core.odm as _foo
 
 config = _foc.load_config()
 annotation_config = _foc.load_annotation_config()
 evaluation_config = _foc.load_evaluation_config()
 app_config = _foc.load_app_config()
 
-_foo.establish_db_conn(config)
+# The old way of doing things, connecting to database on import. If we
+#   REALLY need to do this, for example doc build, we can.
+if "_FIFTYONE_FORCE_DB_CONNECT_ON_IMPORT" in _os.environ:
```

Comment:
If the env var needs to stay, you could turn it into more of a feature by adding it to the [FiftyOneConfig](https://github.com/voxel51/fiftyone/blob/85339fcecf03978435fed6fd94565d1c63f58dd6/fiftyone/core/config.py#L63):

```py
        self._lazy_database_connection = self.parse_bool(
            d,
            "_lazy_database_connection",
            env_var="_FIFTYONE_LAZY_DATABASE_CONNECTION",
            default=True,
        )
```

Private parameters are not included in `print(fo.config)` or `$ fiftyone config` so it's both more cleanly organized and hidden by default.

Having said that, perhaps we should just make this a fully-documented public setting. It would have the benefit of allowing users to switch back to immediate db connections in the event that we discover some issue with lazy connections.